python/guide/module/gluon/hybridize.md

```diff
@@ -1,200 +1,70 @@
-
-Contrary to imperative programming, symbolic programming is usually performed after the computational process has been fully defined. Symbolic programming is used by multiple deep learning frameworks, including Theano and TensorFlow. The process of symbolic programming generally requires the following three steps:
-
-1. Define the computation process.
-2. Compile the computation process into an executable program.
-3. Provide the required inputs and call on the compiled program for execution.
+## Pre-requisite
 
-In the example below, we utilize symbolic programming to re-implement the imperative programming code provided at the beginning of this section.
+Understanding the difference between [Imperative and Symbolic Programming with MXNet](https://d2l.ai/chapter_computational-performance/hybridize.html) and how to create hybridized graphs. 
 
-```{.python .input}
-def add_str():
-    return '''
-def add(a, b):
-    return a + b
-'''
+## Key differences and limitations of hybridized graph
 
-def fancy_func_str():
-    return '''
-def fancy_func(a, b, c, d):
-    e = add(a, b)
-    f = add(c, d)
-    g = add(e, f)
-    return g
-'''
+### Indexing
 
-def evoke_str():
-    return add_str() + fancy_func_str() + '''
-print(fancy_func(1, 2, 3, 4))
-'''
-
-prog = evoke_str()
-print(prog)
-y = compile(prog, '', 'exec')
-exec(y)
+```python
+def hybrid_forward(self, F, x):
+    return x[0,0]
 ```
 
-The three functions defined above will only return the results of the computation process as a string. Finally, the complete computation process is compiled and run using the `compile` function. This leaves more room to optimize computation, since the system is able to view the entire program during its compilation. For example, during compilation, the program can be rewritten as `print((1 + 2) + (3 + 4))` or even directly rewritten as `print(10)`. Apart from reducing the amount of function calls, this process also saves memory.
-
-A comparison of these two programming methods shows that
-
-* imperative programming is easier. When imperative programming is used in Python, the majority of the code is straightforward and easy to write. At the same time, it is easier to debug imperative programming code. This is because it is easier to obtain and print all relevant intermediate variable values, or make use of Python’s built-in debugging tools.
-
-* Symbolic programming is more efficient and easier to port. Symbolic programming makes it easier to better optimize the system during compilation, while also having the ability to port the program into a format independent of Python. This allows the program to be run in a non-Python environment, thus avoiding any potential performance issues related to the Python interpreter.
-
+`TypeError: Symbol only support integer index to fetch i-th output`
 
-## Hybrid programming provides the best of both worlds.
+You cannot use multi-part indexing with Symbols. 
```

Comment:
Should this be `Symbol`s. I'm unsure if we should emphasize the Symbol portion

Reply:
Multi-part indexing like below is unsupported in `Symbol` meaning one has to use work-arounds to accomplish something that can be done simply in Gluon.
```
code
```
**Note: With `Symbol` you have these options ....**